### ISSUE TITLE
Refine generated attribute access modifiers. Originally: Generated Builder.build() method returns concrete implementation not interface type

### ISSUE THREAD

**Issue by Happy-Neko** [immutables/immutables#241]
For following type:
```java
@Value.Immutable
public abstract class Sku {
}
```
generated `ImmutableSku.builder().build()` method returns concrete `ImmutableSku` type. I want it to return abstract `Sku` type.

Am I missing some configuration option?

**Comment by elucash** [comment]
Right, there's style for this `overshadowImplementation`

```java
@Value.Immutable
@Value.Style(overshadowImplementation = true) // style annotation could be put elsewhere
// to not clutter each value definition, see docs for details
interface Value {
}

Value val = ImmutableValue.builder().build(); // returns Value rather than ImmutableValue
```
You can read introduction to styles in http://immutables.github.io/style.html
Thank you!

**Comment by Happy-Neko** [comment]
@elucash 
This works like a charm. Thanks for the help and great library! 

I don't want to spam issues so I'll ask quick unrelated question here: I noticed that generated implementation increases method visibility: `protected abstract Something value()` to `public Something value()` (tested with version 2.1.7).

Is this bug or intended behaviour?

**Comment by elucash** [comment]
Thank you for revealing this! I was not sure about this, then I took a look at it I did noticed some "smartness" there, but, frankly, it do not do what it supposed to, so I revised the logic to handle it in the following way: mirror the same access level as original, except for the case when the style 'visibility' was intentionally set to "PUBLIC"  and it is automatically bumped up to be public. I hope almost no other people would be negatively affected by the change. This change will be release in a next patch release.

Please, feel free to as as many questions as you have!

**Comment by elucash** [comment]
released in 2.1.9
